packages/pwa/app/utils/url.js

```diff
@@ -197,64 +218,84 @@ export const removeQueryParamsFromPath = (path, keys) => {
 }
 
 /**
- * Rebuild the path with locale/site value to the path as url path or url query param
- * based on url config
- * @param {string} url - based url of the output url
+ * Rebuild the path with locale/site values with a given url
+ * The position of those values will based on the url config of current configuration.
+ *
+ * @param {string} relativeUrl - the base relative Url to be reconstructed on
  * @param {object} configValues - object that contains values of url param config
- * @return {string} - an output url
+ * @param {object} opts - options
+ * @return {string} - an output path that has locale and site
  *
  * @example
- * //pwa-kit-config.json
+ * //config/default.json
  * url {
- *    locale: "query_param"
+ *    locale: "query_param",
+ *    site: "path"
  * }
- * buildPathWithUrlConfig('/women/dresses', {locale: 'en-GB'})
  *
- * Returns
- *  /women/dresses?locale=en-GB
- *
- *  @example
- * //pwa-kit-config.json
- * url {
- *    locale: "path"
- * }
- * buildPathWithUrlConfig('/women/dresses', {locale: 'en-GB'})
- *
- * Returns
- *  /en-GB/women/dresses
+ * buildPathWithUrlConfig('/women/dresses', {locale: 'en-GB', site: 'global'})
+ * => /global/women/dresses?locale=en-GB
  *
  */
-export const buildPathWithUrlConfig = (url, configValues = {}) => {
+export const buildPathWithUrlConfig = (relativeUrl, configValues = {}, opts = {}) => {
     const urlConfig = getUrlConfig()
-    if (!urlConfig || !Object.values(urlConfig).length) return url
-    if (!Object.values(configValues).length) return url
-    const queryParams = {}
-    const basePathSegments = []
+    if (!urlConfig) {
+        throw new Error(
+            'url config is required for buildPathWithUrlConfig. Please check your configuration.'
+        )
+    }
+    const {disallowParams = []} = opts
+    if (!Object.values(urlConfig).length) return relativeUrl
+    if (!Object.values(configValues).length) return relativeUrl
+    const [pathname, search] = relativeUrl.split('?')
 
-    const options = ['locale']
+    const params = new URLSearchParams(search)
+    // Remove any disallowed params.
+    if (disallowParams.length) {
+        disallowParams.forEach((param) => {
+            params.delete(param)
+        })
+    }
+
+    const queryParams = {...Object.fromEntries(params)}
+    let basePathSegments = []
+
+    // get the default values for site and locale
+    const showDefaults = urlConfig.showDefaults
+    const defaultSites = [getDefaultSite().id, getDefaultSite().alias]
+    // the default locale values will depend on current site
+    const defaultLocaleId = opts.site?.l10n.defaultLocale
+    const defaultLocaleAlias = opts.site?.l10n.supportedLocales.find(
+        (locale) => locale.alias === defaultLocaleId
+    )
```

Comment:
In fact, the function only needs to know the `defaultLocale` and `defaultLocaleAlias` . I think it is better to make them mandatory instead of passing them into `opts`. Thinking about argment name 🤔

Reply:
To address this, I've changed the interface of this function a little bit. It should take one more arg `defaults` which contains `defaultLocaleRefs` of a current site. See the change [here](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/391/files#diff-f46ab45bbea466e88ac5ab4c430662f8f265bce3754178d40040643bdfc72b3aR275)